lib/src/embedded/isolate_dispatcher.dart

```diff
@@ -98,6 +102,7 @@ class IsolateDispatcher {
     }, onError: (Object error, StackTrace stackTrace) {
       _handleError(error, stackTrace);
     }, onDone: () {
+      _closed = true;
       for (var isolate in _allIsolates) {
```

Comment:
It looks like `_allIsolates` is empty here because we did not add any isolate to it when creating the isolate.

@nex3 I had an attempt to fix closing compiler, but I closed the PR as I see you started to work on this area and it appears that your commit fixed the issue on my side. Maybe you can take a look on the PR and see if there is anything we want to bring here: https://github.com/sass/dart-sass/pull/1985

Reply:
Oops! Must have gotten lost in the shuffle. Thanks for the tip!